### PR TITLE
REPO-5191 Document how to create a custom metadata extractor

### PR DESCRIPTION
**Pull request by alandavis**
Linked to:
* https://github.com/Alfresco/alfresco-community-repo/pull/227
* https://github.com/Alfresco/alfresco-transform-core/pull/316

### CODE REVIEW

File: docs/metadata-extract-embbed.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+Thread-Index=imap:threadIndex
+Message-ID=imap:messageId
+~~~
+As can be seen, the email's metadata for `messageFrom` (if available) will be used to set two properties in the content
+repository (if they exist): `imap:messageFrom`, `cm:originator`. The property names use namespace prefixes specified above.
+
+
+### overwritePolicy
+It is possible to specify if properties in the content repository will be set if the extracted values are not null or if
+the properties are already have a value. By default, `PRAGMATIC` is used. Generally you will not need to change this.
```
Comment (eknizat) [inline docs/metadata-extract-embbed.md:157]: extra 'are' in  'the properties are already have a value'

**Review by eknizat — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,23 @@
   New Features
 </h2>
 
-<h2>        Libraries
-</h2>
+<ul>
+<li>Metadata Extract
+
+The out of the box extraction of metadata is now generally performed asynchronously via a T-Engine connect to the
```
[the remark this answers is not in the document]
Comment (alandavis) [reply CHANGELOG.md:10]: done

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,8 +4,23 @@
...
+repository either as part of the Alfresco Transform Service or as a Local transformer. This provides better security,
+saleability and reliability. The framework used for metadata extraction within the content repository remains,
```
[the remark this answers is not in the document]
Comment (alandavis) [reply CHANGELOG.md:12]: done

File: docs/metadata-extract-embbed.md (hunk 1 above)
Comment (alandavis) [reply docs/metadata-extract-embbed.md:157]: thanks

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (alandavis) [reply CHANGELOG.md:12]: done

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alandavis) [reply CHANGELOG.md:10]: done

File: docs/build_and_release.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+`dependency.alfresco-enterprise-repo.version` use the same SNAPSHOT values.
+
+1. **test**: Java build with unit tests, integration tests and WhiteSource scan.
+2. **docker_latest**: 
+3. **release**: 
+3. **publish**: Artifact deployment to AWS Release bucket.
+
+
+ALL BELOW IS COPIED FROM ANOTHER PROJECT
+
```
Comment (alandavis) [inline docs/build_and_release.md:22]: This .md file (mainly from this point) was copied from alfresco-transform-core project. As a result, the next commit to master after this PR was merged was to tidy it up so this was not included.

**Pull request merged**